src/bblocks/data_importers/imf/dsa.py

```diff
@@ -0,0 +1,215 @@
+
+
+URL: Final[str] = "https://www.imf.org/external/Pubs/ft/dsa/DSAlist.pdf"
+_FOOTNOTE_TRAILER = re.compile(r"\s*\d+/\s*$")
+
+# Columns
+COLS = {
+    1: Fields.country_name,
+    2: "latest_publication",
+    3: "risk_of_debt_distress",
+    5: "debt_sustainability_assessment",
+    6: "joint_with_world_bank",
+    7: "latest_dsa_discussed",
+}
+
+
+def __strip_footnote_trailer(x: str | None) -> str | None:
+    """Strip footnote trailer from string"""
+
+    if not isinstance(x, str):
+        return x
+    return _FOOTNOTE_TRAILER.sub("", x).strip()
+
+
+def __normalise_country_names(df: pd.DataFrame) -> pd.DataFrame:
+    """Normalize country names"""
+
+    if Fields.country_name in df.columns:
+        df[Fields.country_name] = df[Fields.country_name].apply(
+            __strip_footnote_trailer
+        )
+
+    if df[Fields.country_name].isnull().any():
+        raise DataFormattingError("Null values found in country names after cleaning")
+
+    return df
+
+
+def _download_pdf(url: str) -> bytes:
+    """Download PDF"""
+
+    headers = {
+        "User-Agent": "bblocks data importers @ https://data.one.org",
+        "Accept": "application/pdf",
+    }
+
+    try:
+        with httpx.Client(follow_redirects=True, timeout=httpx.Timeout(30.0)) as client:
+            r = client.get(url, headers=headers)
+            r.raise_for_status()
+            return r.content
+
+    except httpx.RequestError as e:
+        raise DataExtractionError(...) from e
+
+    except httpx.HTTPStatusError as e:
+        raise DataExtractionError(...) from e
+
+
+def _pdf_to_df(src: bytes) -> pd.DataFrame:
+    """Extract the single table from the one-page PDF"""
+
+    file = BytesIO(src)
+
+    try:
+        tables = camelot.read_pdf(file, flavor="stream")
+        if len(tables) != 1:
+            raise ValueError("Invalid PDF format. Check PDF")
+
+        return tables[0].df
+
+    except ValueError:
+        raise ValueError("Could not read PDF to a dataframe")
+
+
+def __clean_headers(df: pd.DataFrame) -> pd.DataFrame:
+    """Clean and standardize the DSA table headers."""
+
+    return df.filter(COLS.keys()).rename(columns=COLS)
+
+
+def __normalise_booleans(df: pd.DataFrame, column: str) -> pd.DataFrame:
+    """Normalize boolean columns to True/False."""
+    if column in df.columns:
+        df[column] = df[column].str.lower().eq("yes")
+    return df
+
+
+def __normalise_debt_distress(df: pd.DataFrame) -> pd.DataFrame:
+    """Normalise the "risk of debt distress" column"""
+
+    df["risk_of_debt_distress"] = (
+        df["risk_of_debt_distress"]
+        .str.replace(r"\s+", " ", regex=True)
+        .str.strip()
+        .str.replace("^in debt distress$", "In debt distress", case=False, regex=True)
+        .str.replace("^low$", "Low", case=False, regex=True)
+        .str.replace("^moderate$", "Moderate", case=False, regex=True)
+        .str.replace("^high$", "High", case=False, regex=True)
```

Comment:
I think this could be simplified to 
```python
  df["risk_of_debt_distress"] = (
      df["risk_of_debt_distress"]
      .str.replace(r"\s*\d+/\s*$", "", regex=True) 
      .str.replace(r"\s+", " ", regex=True)
      .str.strip()
      .str.capitalize()
  )
```
but it's optional

Reply:
or instead of `      .str.replace(r"\s*\d+/\s*$", "", regex=True) ` you have a function called `__strip_footnote_trailer`